django-backend/fecfiler/web_services/dot_fec/dot_fec_serializer.py

```diff
@@ -42,6 +43,30 @@ def date_serializer(model_instance, field_name, mapping):
     return date.strftime("%Y%m%d") if date else ""
 
 
+date_formats = [
+    "%Y-%m-%d %H:%M:%S",  # "2024-01-10 00:00:00"
+    "%m/%d/%Y",  # "01/02/2024"
+    "%m/%d/%y",  # "01/02/24"
+    "%m-%d-%Y",  # "01-02-2024"
+    "%m-%d-%y",  # "01-02-24"
+    "%Y-%m-%d",  # "2024-01-10" without time
+]
+
+
+def text_to_date_serializer(model_instance, field_name, mapping):
+    date_string = get_value_from_path(
+        model_instance, mapping.get("path", None) or field_name
+    )
+    for date_format in date_formats:
+        try:
+            # Try parsing with each format in the list
+            date_object = datetime.strptime(date_string, date_format).date()
+            return date_object.strftime("%Y%m%d") if date_object else ""
+        except ValueError:
+            continue  # If it fails, try the next format
+    return date_string
```

Comment:
I feel like we might want to at least log an error here to alert us that a format was entered that we can't match on